tests/test_init.py

```diff
@@ -26,3 +26,208 @@ def test_systemd_present(container):
+            return float(t[:-1])
+        else:
+            raise ValueError("time unit not recognized")
+
+    time = container.connection.run_expect([0], "systemd-analyze time")
+    startup = extract_time(time.stdout, "Startup finished in ")
+    assert startup < 5, "Startup threshold exceeded"
+    target = extract_time(time.stdout, ".target reached after ")
+    assert target < 5, "Reaching systemd target threshold exceeded"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_systemd_nofailed_units(container):
+    """
+    Check that there are not failed systemd units present
+    """
+    output = container.connection.run_expect(
+        [0], "systemctl list-units --state=failed"
+    )
+    assert (
+        "0 loaded units listed" in output.stdout
+    ), "failed systemd units detected"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_systemd_nofailed_units(container):
+    """
+    Check that there are not failed systemd units present
+    """
+    output = container.connection.run_expect(
+        [0], "systemctl list-units --state=failed"
+    )
+    assert (
+        "0 loaded units listed" in output.stdout
+    ), "failed systemd units detected"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_systemd_detect_virt(container, host, container_runtime):
+    """
+    Check that systemd-detect-virt correctly detects the used container runtime
+    """
+    output = container.connection.run_expect([0], "systemd-detect-virt")
+    runtime = container_runtime.runner_binary
+    assert (
+        runtime in output.stdout
+    ), f"systemd-detect-virt failed to detect {runtime}"
+
+
+@pytest.mark.parametrize("container", [INIT_CONTAINER], indirect=True)
+@pytest.mark.skipif(
+    DOCKER_SELECTED
+    and (int(LOCALHOST.package("systemd").version.split(".")[0]) >= 248),
+    reason="Running systemd in docker is broken as of systemd 248, see https://github.com/moby/moby/issues/42275",
+)
+def test_journald(container):
+    """
+    Check that journald behaves correctly
+    """
+
+    # Check that we reached at least the multiuser target
+    journal = container.connection.run_expect([0], "journalctl --boot -e")
```

Comment:
I think you can omit the `-e`, as that only makes a difference when run in interactive mode